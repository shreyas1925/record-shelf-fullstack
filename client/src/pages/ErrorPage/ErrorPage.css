.container--error {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background-color: #ecf0f1;
  text-align: center;
  margin: auto;
  padding: 4em;
}

.container--error img {
  width: 256px;
  height: 225px;
}

.container--error h1 {
  margin-top: 1rem;
  font-size: 35px;
  text-align: center;
}

.container--error h1 > span {
  font-size: 60px;
}

.container--error p {
  margin-top: 1rem;
}

.container--error p.info {
  margin-top: 4em;
  font-size: 12px;
}

.container--error p.info > a {
  text-decoration: none;
  color: rgb(84, 84, 206);
}
