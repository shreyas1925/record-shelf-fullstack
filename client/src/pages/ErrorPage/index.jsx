import React from 'react'
import "./ErrorPage.css"
const ErrorPage = () => {
  return (
    <div className="container--error" data-testid='error-page'>
        <img src="https://i.imgur.com/qIufhof.png" alt='main_image' />
        <h1>
          <span>500</span> <br />
          Internal server error
        </h1>
        <p>We are currently trying to fix the problem.</p>
    </div>
  )
}

export default ErrorPage