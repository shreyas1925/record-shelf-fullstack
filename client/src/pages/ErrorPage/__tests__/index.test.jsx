import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ErrorPage from "..";

describe("Error Page Snapshot", () => {
    it("should render forming a snapshot", () => {
      const { asFragment } = render(<BrowserRouter><ErrorPage/></BrowserRouter>);
      expect(asFragment()).toMatchSnapshot(); 
    });
});

describe("Error Page", () => {
    it("should render page without crashing", () => {
      render(<BrowserRouter><ErrorPage/></BrowserRouter>);
      const errorPageElement = screen.getByTestId("error-page");
      expect(errorPageElement).toBeInTheDocument();
    });

    it("should display text after rendering", () => {
        render(<BrowserRouter><ErrorPage/></BrowserRouter>);
        const errorPageElement = screen.getByText("We are currently trying to fix the problem.");
        expect(errorPageElement).toBeInTheDocument();
      });
});