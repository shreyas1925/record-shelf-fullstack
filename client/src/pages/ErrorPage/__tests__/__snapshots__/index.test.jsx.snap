// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Snapshot should render forming a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container--error"
    data-testid="error-page"
  >
    <img
      alt="main_image"
      src="https://i.imgur.com/qIufhof.png"
    />
    <h1>
      <span>
        500
      </span>
       
      <br />
      Internal server error
    </h1>
    <p>
      We are currently trying to fix the problem.
    </p>
  </div>
</DocumentFragment>
`;
